File: src/tab.cpp
// src/tab.cpp - Basic tab management system
#include <vector>
#include <string>
#include <memory>
#include <iostream>

struct Tab {
    std::string url;
    std::string title;
    int id;
};

static std::vector<Tab> tabs;
static int activeTabIndex = 0;
static int nextTabId = 1;

void CreateTab(const std::string& url = "https://example.com") {
    Tab newTab;
    newTab.url = url;
    newTab.title = "New Tab";
    newTab.id = nextTabId++;
    tabs.push_back(newTab);
    activeTabIndex = tabs.size() - 1;
    std::cout << "Created tab with ID: " << newTab.id << "\n";
    // TODO: Hook into CEF/WebView to actually open the page
}

void CloseTab(int index) {
    if (tabs.empty() || index < 0 || index >= static_cast<int>(tabs.size())) return;
    std::cout << "Closing tab ID: " << tabs[index].id << "\n";
    tabs.erase(tabs.begin() + index);
    if (activeTabIndex >= static_cast<int>(tabs.size())) activeTabIndex = tabs.size() - 1;
}

void SwitchToTab(int index) {
    if (index >= 0 && index < static_cast<int>(tabs.size())) {
        activeTabIndex = index;
        std::cout << "Switched to tab ID: " << tabs[index].id << "\n";
        // TODO: Update CEF/WebView to show that tab
    }
}

const Tab* GetActiveTab() {
    if (tabs.empty()) return nullptr;
    return &tabs[activeTabIndex];
}

const std::vector<Tab>& GetTabs() {
    return tabs;
}

int GetActiveTabIndex() {
    return activeTabIndex;
}

File: src/tab.rs
// src/tab.rs
#[derive(Clone)]
pub struct Tab {
    pub title: String,
    pub url: String,
}

pub struct TabManager {
    pub tabs: Vec<Tab>,
    pub active_index: usize,
}

impl TabManager {
    pub fn new() -> Self {
        Self {
            tabs: vec![Tab {
                title: "Example".into(),
                url: "https://example.com".into(),
            }],
            active_index: 0,
        }
    }

    pub fn switch_to(&mut self, index: usize) {
        if index < self.tabs.len() {
            self.active_index = index;
        }
    }

    pub fn active(&self) -> &Tab {
        &self.tabs[self.active_index]
    }
}
